src/main/scala/firrtl/Emitter.scala

```diff
@@ -751,13 +754,23 @@ class VerilogEmitter extends SeqTransform with Emitter {
     passes.VerilogPrep)
 
   def emit(state: CircuitState, writer: Writer): Unit = {
-    writer.write(preamble)
 
     val circuit = runTransforms(state).circuit
     val moduleMap = circuit.modules.map(m => m.name -> m).toMap
-    circuit.modules.foreach {
-      case m: Module => emit_verilog(m, moduleMap)(writer)
-      case _: ExtModule => // do nothing
+    val (hasRANDOMIZE, emissions) = circuit.modules.foldLeft((false, Seq[String]())) { (trackers, m) =>
```

Comment:
All of the parenthesis on the foldleft is one reason why I prefer using `.empty`, instead this could be:
```scala
val (hasRANDOMIZE, emissions) = circuit.modules.foldLeft((false, Seq.empty[String])) { // ...
```
that's just my opinion though